tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py

```diff
@@ -124,10 +124,13 @@ def build(self, input_shape):
 
     clusterable_weights = self.layer.get_clusterable_weights()
 
-    # Map automatically assigned TF variable name (e.g. 'dense/kernel:0') to provided human readable name 
+    # Map automatically assigned TF variable name (e.g. 'dense/kernel:0') to provided human readable name
     # (e.g. as in Dense(10).kernel)
     clusterable_weights_to_variables = {}
 
+    # To restore the weight array after clustering is done
+    self.weights_num = len(self.layer.weights)
```

Comment:
self.weights_num seems to be only used in _strip_clustering_wrapper.

Just curious: is there a reason not to just directly use self.layer.weights in _strip_clustering_wrapper and remove this self.weights_num attribute?

Furthermore, you might want a check in strip_clustering to make sure the model is built, if that's the assumption.

Reply:
> is there a reason not to just directly use self.layer.weights in _strip_clustering_wrapper and remove this self.weights_num attribute?

Now that you point it out, I believe we should indeed be able to remove this attribute altogether and refer directly to `len(self.layer.weights)` in strip_clustering -- provided that the model is built.

I will apply these changes today and upload a patch as soon as possible.

Reply:
> Just curious: is there a reason not to just directly use self.layer.weights in _strip_clustering_wrapper and remove this self.weights_num attribute?

The original weights from `layer.weights` get removed, therefore we can no longer access them in `strip_clustering()`. On the other hand, we can refer to `len(layer.restore)`, which should coincide with the number of original weights.

Reply:
> Just curious: is there a reason not to just directly use self.layer.weights in _strip_clustering_wrapper and remove this self.weights_num attribute?

Good point, we should indeed be able to get rid of `self.weights_num` and refer to `len(layer.weights)` or `len(layer.restore)` in strip_clustering().